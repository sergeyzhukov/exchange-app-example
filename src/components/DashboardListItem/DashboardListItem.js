import React from 'react'
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native'
import { formatCurrency } from '../../utils/formatters'

export default ({ code, balance, nativeSymbol, name, onPress }) => (
  <TouchableOpacity style={styles.container} onPress={onPress}>
    <View style={styles.icon} />
    <View style={styles.textContainer}>
      <Text style={styles.code}>{code}</Text>
      <Text style={styles.name} numberOfLines={1}>
        {name}
      </Text>
    </View>
    <Text style={styles.amount}>
      {formatCurrency(balance)} {nativeSymbol}
    </Text>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: 64,
    paddingHorizontal: 16,
    alignItems: 'center',
    flexDirection: 'row',
  },
  icon: {
    width: 40,
    height: 40,
    borderRadius: 20,
    marginRight: 12,
    backgroundColor: '#efefef',
  },
  textContainer: {
    flex: 1,
  },
  code: {
    fontSize: 16,
    fontWeight: '400',
    color: '#000000',
  },
  name: {
    marginTop: 4,
    fontSize: 12,
    color: '#747474',
  },
  amount: {
    fontSize: 18,
    fontWeight: '500',
    color: 'black',
  },
})
