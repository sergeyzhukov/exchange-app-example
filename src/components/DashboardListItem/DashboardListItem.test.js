import React from 'react'
import renderer from 'react-test-renderer'
import DashboardListItem from './DashboardListItem'

test('renders correctly', () => {
  const component = <DashboardListItem />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})
