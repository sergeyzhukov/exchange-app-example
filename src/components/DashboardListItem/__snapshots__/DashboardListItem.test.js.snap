// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 64,
      "opacity": 1,
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#efefef",
        "borderRadius": 20,
        "height": 40,
        "marginRight": 12,
        "width": 40,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    />
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#747474",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 18,
        "fontWeight": "500",
      }
    }
  >
    NaN
     
  </Text>
</View>
`;
